File: python/yl09.py
print("Kolmnurga liik")

sides = []
for i in range(3):
    sides.append(float(input(f"sisesta {1+i}. külg: ")))
    if sides[-1] < 0:
        print("Ükski külg ei saa negatiivse pikkusega olla!")
        quit()

sides.sort()
a, b, c = sides

if a + b < c:
    print("See küll pole päris kolmnurk! (lühemad küljed ei ulatu kokku!)")
    quit()

print(
    "See kolmnurk on " + (
        "võrdkülgne" if a == c else
        "võrdhaarne" if a == b or b == c else
        "erikülgne"
    )
)

File: lua/yl09.lua
--------------------------------------------------------------------------------
-- triangle type
--------------------------------------------------------------------------------

print "Kolmnurga liik"

local sides = {}
for i = 1, 3 do
	io.write(("sisesta %i. külg: \t"):format(i))
	table.insert(sides, io.read"*number")
	if sides[i] < 0 then
		print "Ükski külg ei saa negatiivse pikkusega olla!"
		return
	end
end

table.sort(sides)
local a, b, c = table.unpack(sides)

if a + b < c then
	print "See küll pole päris kolmnurk! (lühemad küljed ei ulatu kokku!)"
	return
end

print(("See kolmnurk on %s."):format(
	   (a == b and b == c) and "võrdkülgne"
	or (a == b or b == c)  and "võrdhaarne"
	or "erikülgne"
))

--------------------------------------------------------------------------------
